### ISSUE TITLE
coredump

### ISSUE THREAD

**Issue by ilf** [lfos/calcurse#325]
```
% calcurse --version
calcurse 4.7.0 -- text-based organizer
```

```
% coredumpctl info 203147
           PID: 203147 (calcurse)
           UID: 1023 (user)
           GID: 1023 (user)
        Signal: 6 (ABRT)
     Timestamp: Sun 2020-11-01 19:43:32 CET (34min ago)
  Command Line: calcurse
    Executable: /usr/bin/calcurse
 Control Group: /user.slice/user-1000.slice/session-1.scope
          Unit: session-1.scope
         Slice: user-1000.slice
       Session: 1
     Owner UID: 1000 (user)
       Storage: /var/lib/systemd/coredump/core.calcurse.1000.54c47a0e81e445059d34e430a135bc02.203147.1604256212000000.zst
       Message: Process 203147 (calcurse) of user 1000 dumped core.
                
                Stack trace of thread 203147:
                #0  0x00007f4de0e1d615 raise (libc.so.6 + 0x3d615)
                #1  0x00007f4de0e06862 abort (libc.so.6 + 0x26862)
                #2  0x00007f4de0e5f5e8 __libc_message (libc.so.6 + 0x7f5e8)
                #3  0x00007f4de0e6727a malloc_printerr (libc.so.6 + 0x8727a)
                #4  0x00007f4de0e68d4c _int_free (libc.so.6 + 0x88d4c)
                #5  0x000055e4d906199e n/a (calcurse + 0x1299e)
                #6  0x000055e4d9081ee3 n/a (calcurse + 0x32ee3)
                #7  0x000055e4d9088d2e n/a (calcurse + 0x39d2e)
                #8  0x00007f4de0fbd0f0 __restore_rt (libpthread.so.0 + 0x140f0)
                #9  0x00007f4de0ea7c51 clock_nanosleep@@GLIBC_2.17 (libc.so.6 + 0xc7c51)
                #10 0x00007f4de0ead137 __nanosleep (libc.so.6 + 0xcd137)
                #11 0x00007f4de0ead06e sleep (libc.so.6 + 0xcd06e)
                #12 0x000055e4d90852fa n/a (calcurse + 0x362fa)
                #13 0x000055e4d9089281 n/a (calcurse + 0x3a281)
                #14 0x000055e4d908216e n/a (calcurse + 0x3316e)
                #15 0x000055e4d905509d n/a (calcurse + 0x609d)
                #16 0x00007f4de0e08152 __libc_start_main (libc.so.6 + 0x28152)
                #17 0x000055e4d9055e9e n/a (calcurse + 0x6e9e)
```

**Comment by lfos** [comment]
Thanks for the report, ilf! Can you share some more details on how to reproduce? Is this a binary package or built from source yourself? What's the set of parameters (or the set of actions in interactive mode) that are causing this? Does this also happen with empty configuration and items file? If not, can you try to generate a minimal configuration to reproduce?

**Comment by ilf** [comment]
It's the [Arch package](https://www.archlinux.org/packages/community/x86_64/calcurse/).

This is the config:

```
% cat .calcurse/conf
appearance.calendarview=monthly
appearance.compactpanels=no
appearance.defaultpanel=calendar
appearance.layout=1
appearance.headerline=yes
appearance.eventseparator=yes
appearance.dayseparator=yes
appearance.emptyline=yes
appearance.notifybar=yes
appearance.sidebarwidth=1
appearance.theme=blue on default
appearance.todoview=hide-completed
appearance.headingpos=right-justified
daemon.enable=yes
daemon.log=no
format.inputdate=4
format.notifydate=%F
format.notifytime=%T
format.outputdate=%Y-%m-%d
format.dayheading=%A - %d %B %Y
general.autogc=no
general.autosave=yes
general.confirmdelete=yes
general.confirmquit=yes
general.firstdayofweek=monday
general.multipledays=yes
general.periodicsave=60
general.systemevents=yes
general.systemdialogs=no
notification.command=calcurse --next | mail -s "[calendar] upcoming event" user
notification.notifyall=all
notification.warning=300
appearance.emptyday=--
```

Unfortunately, I do not have the resources to provide complete steps-to-reproduce ATM.

I have repeadedly experienced the interactive mode "freeze", especially in the time between notification for an upcoming event and the event itself, sometimes even after the event. I'm not sure if this is related, though. (And I have'd had time to properly debug and provide steps-to-reproduce for that either, sorry.) But I wanted to let you know about the coredump, I assume this should never happen.

Let me know if I can do anything else to help.

Thanks for your work!

**Comment by lfos** [comment]
Thanks for the additional details; that already helps a lot. Are you using any hooks?

**Comment by ilf** [comment]
No, I am not using hooks, `.calcurse/hooks/` is empty.

**Comment by ilf** [comment]
via #326

> Have you tried running calcurse in an ordinary terminal (not tmux)?

I have just tried calcurse directly in a terminal, without tmux. Same thing.

> How exactly do you invoke calcurse (complete command line).

```
% which calcurse
/usr/bin/calcurse

% calcurse
```

**Comment by lfos** [comment]
Is this still happening? Were you able to isolate the root cause in the meantime?

**Comment by ilf** [comment]
I haven't experienced this in a long time. I assume it has something to do with #326, for which I am now using a work-around.

**Comment by lfos** [comment]
Thanks for confirming. The `pu` branch contains some changes that should fix #326 without the need for a workaround, will likely be merged into master soon.
